provider/plugin/plugin.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+	"sigs.k8s.io/external-dns/provider"
+)
+
+type PluginProvider struct {
+	provider.BaseProvider
```

Comment:
BaseProvider has no domainfilter feature as far as I see GetDomainFilter. Do you intend that the 3rd party provider filters for us?
If it would be a generic feature we can add it here which frees the 3rd party providers from that code.

Reply:
Good point, I need to think about how this will be 🤔

Reply:
Looking at this, the issue is really that the provider interface has `GetDomainFilter() endpoint.DomainFilterInterface` which returns an object that does the filter operation and that wouldn't really make sense to do over HTTP. 

The other problem, is also that we need to add the two other functions:

- `PropertyValuesEqual`
- `AdjustEndpoints`

My thinking is that we can add those two, but `GetDomainFilter`'s logic needs to be implemented by the providers. WDYT @szuecs ?

Reply:
As discussed in person: 
- PropertyValuesEqual
- AdjustEndpoints
will be part of the webhook interface.

DomainFilter somehow needs to be done by the provider itself and is out of scope right now.
Maybe we can think about a list of regexp domain filter in the plugin provider and expose config via cli or configmap, not sure if this would be sufficient.